### PR TITLE
Bump rust to 1.37.0

### PR DESCRIPTION
**Pull request by dagurval**
The minimal supported rust version in next ElectrsCash release will be 1.34.0. Bitcoin Unlimited gitian build currently has 1.32.0.

This bumps the rust version in the gitian build to the latest stable release, which is 1.37.0.

### CODE REVIEW

File: src/Makefile.am [hunk 1]
```diff
@@ -181,6 +181,7 @@ BITCOIN_CORE_H = \
   random.h \
   reverse_iterator.h \
   reverselock.h \
+  rpc/blockchain.h \
```
Comment (sickpig) [inline src/Makefile.am:184]: Seems unrelated, no?

Comment (dagurval) [reply src/Makefile.am:184]: Yes, but needed to test gitian build, which is broken without this header

**Comment by dagurval** [comment]
Result of gitian build (`./bin/gbuild -j 16 -m 10000 --url bitcoin=https://github.com/dagurval/BitcoinUnlimited.git --commit bitcoin=bump-rust ../bu/contrib/gitian-descriptors/gitian-linux.yml`)

```
1be067b12a207494d7e75cefbd6d5e49a7d9f0800b4cf65352334928282b536d  BUcash-1.6.0.1-linux32.tar.gz
2744c8505014f7249e847c8d0a5920c1dd6802af230341ede7b359a290d81142  BUcash-1.6.0.1-linux64.tar.gz
91a5daeeecc355cfd28908959d085620c18dacedf7a2371d59b8a39243647920  bitcoin-qt-i686-pc-linux-gnu.debug
705871a9006a2d096cb3805e70785c384ba827555bf9cb2a5aa48970bae7437a  bitcoin-qt-x86_64-linux-gnu.debug
6701e86e83227f83164d22bf8ba48e8a552cb7c3b6ab1690cd604c493f3eb862  bitcoind-i686-pc-linux-gnu.debug
9b2c4a115ad2ee1e59fbf1686fff9df14ee513dbb4bb1c9547de5946ff7a1327  bitcoind-x86_64-linux-gnu.debug
2582e73bd263e0be820732ecf9822f7c6af0dcfacd6e0b790f892a25a8e199d1  src/BUcash-1.6.0.1.tar.gz
37c4bc6c29fcf7ba8d0ca857d8f36279ff592b868a7772708957f60561ea8321  BUcash-linux-1.6-res.yml
```
